### PR TITLE
feature: S3C-2260 Add unit and functional testing

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Maintenance testing for `UtapiReindex`:
* start Redis Sentinel as a process
* add basic tests for `UtapiReindex` methods connecting to Redis Sentinel

### CODE REVIEW

File: eve/workers/unit_and_feature_tests/redis/sentinel.conf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Replicas are auto-discovered, so you don't need to specify replicas in
+# any way. Sentinel itself will rewrite this configuration file adding
+# the replicas using additional configuration options.
+# Also note that the configuration file is rewritten when a
+# replica is promoted to master.
+#
+# Note: master name should not include special characters or spaces.
+# The valid charset is A-z 0-9 and the three characters ".-_".
+sentinel monitor scality-s3 127.0.0.1 6379 2
```
Comment (ssalaues) [inline eve/workers/unit_and_feature_tests/redis/sentinel.conf]: It's very likely this is silently failing. You have a quorum of `2` set but there is only 1 Redis server.

Comment (bennettbuchanan) [reply eve/workers/unit_and_feature_tests/redis/sentinel.conf]: I agree this should be updated to 1 to be accurate and have pushed that change.

However, if the sentinel can't connect to the Redis server the test will not fail silently regardless of the quorum number. The refused connect will fail with an error:

```
Error: connect ECONNREFUSED 127.0.0.1:6379
```

Comment (bennettbuchanan) [reply eve/workers/unit_and_feature_tests/redis/sentinel.conf]: Thanks, I'm looking into this a bit further to be certain. It's possible that the error does not come from the sentinel though.

Comment (ssalaues) [reply eve/workers/unit_and_feature_tests/redis/sentinel.conf]: Hmm that's interesting. Just let me know you need help debugging the environment. Sometimes peaking through the Berte/Eve curtains can be very useful.

Comment (bennettbuchanan) [reply eve/workers/unit_and_feature_tests/redis/sentinel.conf]: I just double checked this, and yes, the sentinel will respond with an error regardless of the quorum number when the client emits the 'error' event [here](https://github.com/scality/utapi/blob/development/8.0/lib/UtapiReindex.js#L165):

```
{ Error: connect ECONNREFUSED 127.0.0.1:6379
    at Object._errnoException (util.js:1026:11)
    at _exceptionWithHostPort (util.js:1049:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1174:14)
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 6379 }
```

Additionally, if the Redis Server is running but there is no Redis Sentinel, the response is as expected:

```
Error: All sentinels are unreachable. Retrying from scratch after 10ms. Last error: Connection is closed.
```

The client which is connecting to the Redis Server directly in the `beforeEach` [here](https://github.com/scality/utapi/pull/294/files#diff-474cc7d7664919efb3ec1d789deb7e21R13) will also emit the same error which is why I thought there was a chance it wasn't coming from the sentinel in fact.

In any case, setting the value to 1 seems more correct to me. I'll let you know if there is a need for debugging the environment. Thanks! 😃

File: tests/functional/testReindex.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    beforeEach(done => {
+        reindex = new UtapiReindex();
+        redis = redisClient({}, log)
+            .on('ready', done)
+            .on('error', done);
+    });
+
+    afterEach(done => {
+        redis.flushdb(done);
```
Comment (JianqinWang) [inline tests/functional/testReindex.js]: maybe a good idea to close the client as well if we're re-instantiating a redisclient for every test

**Review by JianqinWang — APPROVED** [review]

**Review by ssalaues — APPROVED** [review]

File: tests/functional/testReindex.js (hunk 2 above)
Comment (bennettbuchanan) [reply tests/functional/testReindex.js]: Good idea, I've updated that.

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@bert-e approve

**Pull request merged**